### DISCUSSION TITLE
VICI events during Child rekey collisions

### DISCUSSION THREAD

**Discussion by bhatnitish** [strongswan/strongswan#1698] [Q&A] [answered]
Hi,

When subscribing to VICI child rekey events, we see 2 back to back events when there's a child rekey collision. Both events have the same "old" sa information, so tracking the SA progression breaks. I see this was discussed here: https://github.com/strongswan/strongswan/discussions/1234 and looks like the fix (https://github.com/strongswan/strongswan/commit/d0d2336b1bf534dea7ba7c8edc77703940a05076) has not made it into the master or any releases yet. In the time being, could we assume the the child rekey events will always arrive in an ordered fashion - meaning the "first" event will be the SA that'll go away and the "second" rekey event will always be the winning child sa event?

**Comment by tobiasbrunner** [comment, marked as answer]
> and looks like the fix ([d0d2336](https://github.com/strongswan/strongswan/commit/d0d2336b1bf534dea7ba7c8edc77703940a05076)) has not made it into the master or any releases yet.

The fix is included in the [6.0 beta releases](https://www.strongswan.org/download.html). I recommend using that if these events are important to you.

> In the time being, could we assume the the child rekey events will always arrive in an ordered fashion - meaning the "first" event will be the SA that'll go away and the "second" rekey event will always be the winning child sa event?

If there are two events for the same old SA, then yes, the new SA in the first event will be the redundant one the peer is expected to delete (it might already have done so once the second event is triggered).
